src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegment.netstandard.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+
+namespace System.IO.Pipelines
+{
+    internal sealed partial class BufferSegment : ReadOnlySequenceSegment<byte>
+    {
+        public void ResetMemory()
+        {
+            object memoryOwner = _memoryOwner!;
+
+            // Order of below field clears is significant as it clears in a sequential order
+            // https://github.com/dotnet/corefx/pull/35256#issuecomment-462800477
```

Comment:
The linked comment doesn't make it clear what hte purpose is. If it's to get better codegen should there be an issue against that?

Reply:
Is sequential memory access order rather than randomly jumping around

Reply:
Is sequential memory access order rather than randomly jumping around; don't think the CLR memory model allows reording writes so the Jit couldn't improve it without changing the C#?

Reply:
Ah right. Maybe the link doesn't add much - could just be explicit
```
// Order of below field clears is significant as it clears in a sequential order
// which measurably improves perf
```